### ISSUE TITLE
Scheduler becomes unavailable

### ISSUE THREAD

**Issue by martinhoess** [nielsfaber/scheduler-component#198]
So, I've two schedules and until one of the latest HA/Scheduler versions they did run without any problem. At some point the scheduler card did show the error message. I've re downloaded / removed the scheduler component without any luck.  If I remove the storage file then the error message is gone. If I put back the storage file and remove the second schedule, it will work fine. So something is wrong with the second schedule? However if I start with a blank storage and create the two schedules from scratch it will work until I restart HA. The calendar are based on the google calendar integration. Reloading the scheduler integration at a later point doesn't help.

### Checklist
- [X] I checked for similar existing issues (both open and closed) before posting.
- [X] I am running the latest version.

### Expected behavior
Load and show both schedules correctly

### Actual behavior
Scheduler card shows "Could not connect with the scheduler component. It needs to be installed as integration before this card can be used." and second schedule has state "unavailable"
<img src="https://user-images.githubusercontent.com/47973549/166105235-b691ceef-e99a-42b3-9865-9543c98328b2.png" width="200" />

### Steps to Reproduce
see attached scheduler
[scheduler.storage.txt](https://github.com/nielsfaber/scheduler-component/files/8596868/scheduler.storage.txt)

**Comment by BlueSam73** [comment]
I do have exactly same behavior.
Does anyone has solution?

**Comment by nielsfaber** [comment]
I will try and see if I can get the same error with the attached storage file. I will post here later this week

**Comment by nielsfaber** [comment]
So I did some tests and found the following:

1. After using your scheduler.storage file I had the same error
2. Scheduler was unable to calculate the next occurrence for schedule #764ba5. This is also shown in the logs: `WARNING (MainThread) [custom_components.scheduler.timer] failed to calculate next timeslot for schedule 764ba5`. Please check the logs next time, it provides useful info.
3. The reason why the next occurrence cannot be calculated, is because this schedule was configured to end on 2022-04-30 which is in the past. Normally scheduler would jump back to that date and consider that date as next (final) occurrence, and mark the schedule as expired. In this case that date was not a weekday (as was required by the schedule configuration), so scheduler failed to combine those requirements. As an improvement the scheduler should be adapted to roll back to the last working day before 2022-04-30, but this is currently not in place.
4. The card was able to communicate with the scheduler, but could not process the schedule #764ba5 (since no trigger occurence was calculated). When the scheduler card fails to display one or more, it shows the same error as when communication fails. I made some changes to have only an error displayed for the affected schedule (https://github.com/nielsfaber/scheduler-card/commit/c46e28f51a650d492b430f0c3ade20f162a2b6a2).

So in summary, I think we have found a bug and scheduler needs to be adapted to be able to handle cases where the end date is in the past whilst that end date is not in the list of allowed dates.

As an intermediate fix, you could remove the date restriction (set start_date to null and end_date to null) or weekday restriction (set to daily) in the scheduler.storage file (+ restart HA). It should make the schedules visible again in the card.

**Comment by martinhoess** [comment]
Thanks for having a look.

2. Sorry about that. I actually always check them and especially after a HA update. No idea why I've missed that.

Setting the end date to a weekday as workaround works 👍

**Comment by nielsfaber** [comment]
This issue should be fixed in the v3.2.9 release.
Please re-open if you still encounter issues after updating.

**Comment by machejc** [comment]
I am running Raspberry Pi OS bullseye and have seemingly installed HACS and Schedule component correctly and see if appearing in HACS, Integrations. I have also appeared to have installed Scheduler Card correctly and can see it in HACS, Frontend. When I try to add the card to my dashboard I get the error: "Could not connect with the scheduler component. It needs to be installed as integration before this card can be used." in card configuration. What am I missing?

**Comment by machejc** [comment]
Home Assistant 2023.3.6
Frontend 20230309.1 - latest
